vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -548,6 +548,11 @@ func (ep *endpoint) updateDNS(resolvConf []byte) error {
 		return err
 	}
 
+	// Change the perms to 0644 since ioutil.TempFile creates it by default as 0600
+	if err := os.Chmod(tmpResolvFile.Name(), 0644); err != nil {
```

Comment:
No particular reason. Could've used `file.Chmod` as well. But `os.Chmod` does not open the file and call `file.Chmod`. It just calls `chmod` system call while `file.Chmod` calls `fchmod` system call. That's the only difference.